src/main/java/uk/gov/hmcts/reform/sscs/util/PdfRequestUtil.java

```diff
@@ -0,0 +1,189 @@
+        }
+        pdfUrlBuilder.append(".pdf");
+
+        if (isBlank(requestDetails)) {
+            String responseErrorMsg = String.format("Please enter request details to generate a %s document", pdfType.toString().toLowerCase());
+            response.addError(responseErrorMsg);
+            return response;
+        }
+
+        DocumentLink previewDocument = getPreviewDocument(pdfUrlBuilder.toString(), userAuthorisation, generateFile, templateId, title);
+
+        switch (pdfType) {
+            case POST_HEARING:
+                sscsCaseData.getDocumentStaging().setPreviewDocument(previewDocument);
+                break;
+            case POSTPONEMENT:
+                sscsCaseData.getPostponementRequest().setPostponementPreviewDocument(previewDocument);
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported event type for processRequestPdfAndSetPreviewDocument: " + pdfType);
+        }
+
+        return response;
+    }
+
+    private static void handlePostponement(SscsCaseData sscsCaseData) {
+        PostponementRequest postponementRequest = sscsCaseData.getPostponementRequest();
+        requestDetails = postponementRequest.getPostponementRequestDetails();
+        String hearingVenue = postponementRequest.getPostponementRequestHearingVenue();
+        LocalDate hearingDate = LocalDateTime.parse(postponementRequest.getPostponementRequestHearingDateAndTime()).toLocalDate();
+        additionalRequestDetails.append("Date of Hearing: ")
+            .append(hearingDate.format(DATE_TIME_FORMATTER))
+            .append("\n")
+            .append("Hearing Venue: ")
+            .append(hearingVenue)
+            .append("\n")
+            .append("Reason for Postponement Request: ")
+            .append(requestDetails)
+            .append("\n");
+        title = "Postponement Request from FTA";
+    }
+
+    private static void handlePostHearing(SscsCaseData sscsCaseData) {
+        requestDetails = sscsCaseData.getDocumentGeneration().getBodyContent();
+        LocalDate issueFinalDecisionDate = sscsCaseData.getIssueFinalDecisionDate();
+        if (issueFinalDecisionDate == null) {
+            throw new IllegalArgumentException("issueFinalDecisionDate unexpectedly null for caseId: " + sscsCaseData.getCcdCaseId());
+        }
+        String postHearingRequestType = sscsCaseData.getPostHearing().getRequestType().getDescriptionEn();
+        additionalRequestDetails.append("Date of decision issued: ")
+            .append(issueFinalDecisionDate.format(DATE_TIME_FORMATTER))
+            .append("\n")
+            .append("Reason for ")
+            .append(postHearingRequestType)
+            .append(" application: ")
+            .append(requestDetails)
+            .append("\n");
+        title = String.format("%s Application from %s", postHearingRequestType, "FTA"); // TODO SSCS-10759 make dynamic for the uploading party
```

Comment:
Same as above, can remove for now as 10759 likely doesn't require this code.

Reply:
removed